[fix] resolve build error for make start without frontend dist

## 问题描述
执行 `make start` 时报错：

```
server/router/frontend/frontend_prod.go:9:12: pattern dist/*: no matching files found
make: *** [build] Error 1
```

## 根本原因

1. `make start` 和 `make restart` 错误地依赖了 `build` 目标
2. `build` 目标没有使用 `-tags=noui`，导致需要 `dist/` 目录
3. `scripts/dev.sh` 使用 `go run` 但没有添加 `-tags=noui` 标志

## 解决方案

- 移除 `start` 和 `restart` 对 `build` 的依赖
- 在 `scripts/dev.sh` 中添加 `-tags=noui` 到 `go run` 命令
- 开发环境使用 `frontend.go` (noui) 而非 `frontend_prod.go`

## 验收标准

- [ ] `make start` 正常启动所有服务
- [ ] `make restart` 正常重启所有服务
- [ ] 后端日志无 embed 相关错误

--- a/Makefile
+++ b/Makefile
@@ -94,13 +94,13 @@ dev: run ## Alias for run
 web: ## 启动前端开发服务器
 	@cd $(WEB_DIR) && pnpm dev
 
-start: build ## 一键启动所有服务 (自动构建最新版本)
+start: ## 一键启动所有服务 (使用 go run 开发模式)
 	@$(SCRIPT_DIR)/dev.sh start
 
 stop: ## 一键停止所有服务
 	@$(SCRIPT_DIR)/dev.sh stop
 
-restart: build ## 重启所有服务 (自动构建最新版本)
+restart: ## 重启所有服务 (使用 go run 开发模式)
 	@$(SCRIPT_DIR)/dev.sh restart
 
 status: ## 查看所有服务状态

--- a/scripts/dev.sh
+++ b/scripts/dev.sh
@@ -194,7 +194,7 @@ start_backend() {
     load_env
 
     # 启动后端（后台运行）
-    nohup go run ./cmd/divinesense --mode dev --port $BACKEND_PORT \
+    nohup go run -tags=noui ./cmd/divinesense --mode dev --port $BACKEND_PORT \
         > "$BACKEND_LOG" 2>&1 &
 
     local pid=$!

